### PR TITLE
docker: Add arm64 builds to Dockerfile and CI

### PR DESCRIPTION
**Pull request by nbuchwitz**
Follow-up of @tboehler1 's #505 as suggested by @obbardc 

Convert Dockerfile so multiple target architecturebuilds are possible. Also modify the CI pipeline, in order to built a single multi-platform container.

Note that at the current time-being docker cannot export multi-platform containers without an intermediate registry. Thus only the amd64 image is exported for the test stage. Nevertheless the arm64 image is built in the pipeline and also published later to the registry.

If the go-debos project has access to GitHub large runners, the pipeline could be reworked in order to build the container on a native arm runner. This would bring a significant speed improvement and also would allow for testing the arm container. But for know this solution should be sufficient as it tests the build on both architectures and runs the test suite on amd64.

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (obbardc) [inline .github/workflows/ci.yaml:78]: In future, could we work around this with a registry in the runner ? We should really run tests on both docker images :-)

File: .github/workflows/ci.yaml
Comment (obbardc) [inline .github/workflows/ci.yaml:268]: expand to a variable e.g `DOCKER_PLATFORMS` ?

File: docker/Dockerfile
Comment (obbardc) [inline docker/Dockerfile:2]: I think this needs to be more descriptive e.g `IMAGE_PREFIX_ARCH`

File: docker/Dockerfile
Comment (obbardc) [inline docker/Dockerfile:51]: spacing to be consistent with these two lines :-)

**Review by obbardc — CHANGES_REQUESTED** [review]

**Comment by obbardc** [comment]
We only have amd64 runners, I think.

File: .github/workflows/ci.yaml
Comment (nbuchwitz) [reply .github/workflows/ci.yaml]: Makes sense. I moved the supported platforms into `DOCKER_TARGET_PLATFORMS`

File: docker/Dockerfile
Comment (nbuchwitz) [reply docker/Dockerfile]: Actually this isn't really needed as we use buildkit and a base image which already has support for the target platforms. Thus I have removed the `ARCH` prefix in the latest series. Damn musle memory from times before multi platform support in Docker :-)

File: .github/workflows/ci.yaml
Comment (nbuchwitz) [reply .github/workflows/ci.yaml:79]: Yes this would resolve the issue. It also makes it really easy for a reviewer to test a PR's content within a container. With a proper retention policy the containers can be cleaned up afterwards. Alternatively a hook on PR close/merge could do the cleanup.

**Comment by nbuchwitz** [comment]
All of the review feedback should have been addressed with the latest patch series. I was also able to figure out, why the unit tests failed earlier. Turns out this was unrelated to my modifications and more an issue with GitHub switching over from v1 to v2 compose api. Fixed this in another commit. 

CI works as expected (in my fork): https://github.com/nbuchwitz/debos/actions/runs/10537327064

File: .github/workflows/ci.yaml
Comment (obbardc) [reply .github/workflows/ci.yaml:79]: I created https://github.com/go-debos/debos/issues/514 to track the unit tests. I don't think we should block on that for merging this PR.

**Review by obbardc — APPROVED** [review]

**Comment by nbuchwitz** [comment]
The failed job in https://github.com/go-debos/debos/actions/runs/10537327383/job/29218707965?pr=513#step:6:3778 looks like the arch server was unresponsive during curl run. Could you please restart the job?

**Comment by obbardc** [comment]
> The failed job in https://github.com/go-debos/debos/actions/runs/10537327383/job/29218707965?pr=513#step:6:3778 looks like the arch server was unresponsive during curl run. Could you please restart the job?

Sorry @nbuchwitz I haven't been able to dedicate any time to debos recently. I can't seem to run the job. Can you navigate to 
 https://github.com/nbuchwitz/debos/actions/workflows/ci.yaml and run the workflow manually for your branch ?

Thank you!

**Comment by nbuchwitz** [comment]
I've ran the CI as requested here: https://github.com/nbuchwitz/debos/actions/runs/11581471675

**Comment by obbardc** [comment]
@nbuchwitz it still shows as not passing on the PR and I can't see how to force it. Perhaps you have to force-push a new commit ?

**Comment by nbuchwitz** [comment]
The pipeline status in this pull request in inherited from merge request job and not the job which ran in my fork. Thus the job needs to be retriggered here. Just did a force push, guess you will have to start the job

**Comment by obbardc** [comment]
> The pipeline status in this pull request in inherited from merge request job and not the job which ran in my fork. Thus the job needs to be retriggered here. Just did a force push, guess you will have to start the job

I've approved it, it's running! Apologies for all of that.

**Comment by nbuchwitz** [comment]
Fingers crossed :smile:

**Comment by nbuchwitz** [comment]
All green! Looking forward to point our build servers to the native arm64 releases after merge ! :partying_face:

**Comment by obbardc** [comment]
Happy with this PR. Thanks for your contribution @nbuchwitz

**Pull request merged**
